### ISSUE TITLE
`core:container/bit_array` (presumably) generates faulty code on Linux

### ISSUE THREAD

**Issue by ap29600** [odin-lang/Odin#1468]
## Context

```
Odin: dev-2022-02:abb26e0b
OS:   Arch Linux, Linux 5.16.2-arch1-1
CPU:  AMD Ryzen 5 2500U with Radeon Vega Mobile Gfx
RAM:  7652 MiB
```

## Expected Behavior
Creating a bit_array and setting some bits results in only those bits being set.

## Current Behavior
Some extra bits are set when compiling and running on Linux, deterministically with respect to the bits being explicitly set.
for example setting the bits: `568, 9420, 3294, 9446, 9443, 3711, 6730, 5147, 6683, 603, 3845` 
additionally  sets the bits: `640, 644, 651, 657`.

### Steps to Reproduce

compile and run the following code on Linux:

```odin
package bug
import "core:math/rand"
import bit "core:container/bit_array"
import "core:fmt"

sz :: 10000
TEST := []int{568, 9420, 3294, 9446, 9443, 3711, 6730, 5147, 6683, 603, 3845}

main :: proc () {
    b := bit.create(sz)
    for i in TEST {
        bit.set(&b, i)
        fmt.println(i)
    }

    count_b := 0
    fmt.println("=======")
    for i in 0..=sz {
    if val, ok := bit.get(&b, i); val && ok {
        fmt.println(i)
        count_b += 1
    }
    }
    fmt.println("=======")
    fmt.println(count_b)
}
```

the result is:

```
568
9420
3294
9446
9443
3711
6730
5147
6683
603
3845
=======
568
603
640
644
651
657
3294
3711
3845
5147
6683
6730
9420
9443
9446
=======
15
```

**Comment by Kelimion** [comment]
Confirmed on:
```
Odin: dev-2022-02:abb26e0b
OS:   Ubuntu 20.04.3 LTS, Linux 5.4.72-microsoft-standard-WSL2
CPU:  AMD Ryzen 7 1800X Eight-Core Processor
RAM:  51363 MiB
```

**Comment by Kelimion** [comment]
Fixed via #1376.
